src/main/java/uk/gov/companieshouse/api/accounts/validation/CreditorsWithinOneYearValidator.java

```diff
@@ -0,0 +1,191 @@
+import uk.gov.companieshouse.api.accounts.transaction.Transaction;
+
+@Component
+public class CreditorsWithinOneYearValidator extends BaseValidator {
+
+    @Value("${invalid.note}")
+    private String invalidNote;
+
+    private static final String CREDITORS_WITHIN_PATH = "$.creditors_within_one_year";
+    private static final String CREDITORS_WITHIN_CURRENT_PERIOD_PATH = CREDITORS_WITHIN_PATH +
+            ".current_period";
+    private static final String CREDITORS_WITHIN_PREVIOUS_PERIOD_PATH = CREDITORS_WITHIN_PATH +
+            ".previous_period";
+    private static final String CREDITORS_WITHIN_CURRENT_PERIOD_TOTAL_PATH =
+            CREDITORS_WITHIN_CURRENT_PERIOD_PATH + ".total";
+    private static final String CREDITORS_WITHIN_PREVIOUS_PERIOD_TOTAL_PATH =
+            CREDITORS_WITHIN_PREVIOUS_PERIOD_PATH + ".total";
+    private static final String CREDITORS_WITHIN_CURRENT_PERIOD_DETAILS_PATH =
+            CREDITORS_WITHIN_CURRENT_PERIOD_PATH + ".details";
+
+    private CompanyService companyService;
+
+    @Autowired
+    public CreditorsWithinOneYearValidator(CompanyService companyService) {
+        this.companyService = companyService;
+    }
+
+    public Errors validateCreditorsWithinOneYear(@Valid CreditorsWithinOneYear creditorsWithinOneYear, Transaction transaction) throws DataException {
+
+        Errors errors = new Errors();
+
+        if (creditorsWithinOneYear != null) {
+
+            if (creditorsWithinOneYear.getCurrentPeriod() != null) {
+
+                CurrentPeriod creditorsCurrentPeriod = creditorsWithinOneYear.getCurrentPeriod();
+
+                validateCurrentPeriod(creditorsCurrentPeriod, errors);
+            }
+
+            if (creditorsWithinOneYear.getPreviousPeriod() != null) {
+
+                PreviousPeriod creditorsPreviousPeriod = creditorsWithinOneYear.getPreviousPeriod();
+
+                try {
+
+                    if (companyService.isMultipleYearFiler(transaction)) {
+
+                        validatePreviousPeriod(creditorsPreviousPeriod, errors);
+
+                    } else {
+
+                        validateInconsistentFiling(creditorsPreviousPeriod, errors);
+                    }
+                } catch (ServiceException e) {
+                    throw new DataException(e.getMessage(), e);
+                }
+            }
+        }
+        return errors;
+    }
+
+    private void validateInconsistentFiling(@Valid PreviousPeriod creditorsPreviousPeriod,
+            Errors errors) {
+        if (creditorsPreviousPeriod.getTotal() != null) {
+
+            addInconsistentDataError(errors,
+                    CREDITORS_WITHIN_PREVIOUS_PERIOD_TOTAL_PATH);
+        }
+    }
+
+    private void validatePreviousPeriod(@Valid PreviousPeriod creditorsPreviousPeriod,
+            Errors errors) {
+        validatePreviousTotalCalculationCorrect(creditorsPreviousPeriod, errors);
+        validatePreviousTotalProvidedIfValuesEntered(creditorsPreviousPeriod, errors);
```

Comment:
The logic in this method is ordered in the opposite way to what I would expect, personally.
Would it make more sense to...
1) check total is not null if other fields are provided
2) if errors are present, return - there isn't anything further to validate
3) else, validate the totals?